### ISSUE TITLE
AOT Eager issue w/ QAT

### ISSUE THREAD

**Issue by mlazos** [pytorch/pytorch#161569]
### 🐛 Describe the bug

This is for an internal RL usecase (ping me internally for more info). With https://github.com/pytorch/pytorch/pull/152934 which begins inlining some TorchAO modules, a bug is uncovered with resnet50_quantized_qat. 

`python benchmarks/dynamo/torchbench.py --accuracy --float32 -d cpu -n10 --training --only resnet50_quantized_qat --cold-start-latency --backend=aot_eager`

will raise https://gist.github.com/mlazos/aa4e0515eba84a2b9c5792a8632a9aeb

There is a function with a known buggy meta https://github.com/pytorch/pytorch/issues/159192 involved which may be part of the issue, but fwiw this code does run in eager. I filed a bug in the AO repo as well. 

NB: This repros on all devices, running on cpu is the simplest though. 

### Error logs

https://gist.github.com/mlazos/aa4e0515eba84a2b9c5792a8632a9aeb

### Versions

nightly cd87f3029582cedb3b88747a3bd7d200b05c1138

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @chauhang @penguinwu @bdhirsh

**Comment by anijain2305** [comment]
Triage notes: workaround exists for inductor, but not for aot_eager. Would be good to get it done in next 1-2 months. @bdhirsh to take a fist look.

**Comment by aorenste** [comment]
This issue no longer reproduces on PyTorch 2.12.0a0+git89d1757.

The triggering change (#152934, which inlined into AO quantization modules) was reverted in #161567 on Aug 27, 2025. This prevents dynamo from inlining into the QAT modules, so the buggy meta kernel for `_fused_moving_avg_obs_fq_helper` is never invoked during AOT tracing.

Note that the underlying meta kernel bug (#159192) is still open and would need to be fixed before re-landing the inlining change.

(Analysis done by Claude.)
